plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreAttribute.java

```diff
@@ -434,7 +436,9 @@ public boolean allowCustomValue()
                 log.debug("Can't get data types from database schemas", e);
                 types.addAll(column.getDatabase().getLocalDataTypes());
             }
-            return types.toArray(new PostgreDataType[0]);
+            PostgreDataType[] notArrayTypes = types.stream().filter(type -> type.getTypeID() != Types.ARRAY).sorted(Comparator.comparing(JDBCDataType::getTypeName)).toArray(PostgreDataType[]::new);
+            PostgreDataType[] arrayTypes = types.stream().filter(type -> type.getTypeID() == Types.ARRAY).sorted(Comparator.comparing(JDBCDataType::getTypeName)).toArray(PostgreDataType[]::new);
+            return ArrayUtils.concatArrays(notArrayTypes, arrayTypes);
```

Comment:
Instead of this madness, you can use this instead:

```java
return types.stream()
    .sorted(Comparator
        .comparing((DBSTypedObject type) -> type.getTypeID() != Types.ARRAY)
        .thenComparing(DBSTypedObject::getTypeName))
    .toArray(PostgreDataType[]::new);
```

Reply:
Since `types` is a List, there is no need to create a stream, you can sort the list itself.